src/rustdoc.md

```diff
@@ -36,30 +36,174 @@ does is call the `main()` that's in this crate's `lib.rs`, though.)
+* The bits specific to using rustdoc as a test harness are in
+  `src/librustdoc/doctest.rs`.
+* The Markdown renderer is loaded up in `src/librustdoc/html/markdown.rs`, including functions
   for extracting doctests from a given block of Markdown.
-* The tests on rustdoc *output* are located in `src/test/rustdoc`, where
+* The tests on the structure of rustdoc HTML output are located in `src/test/rustdoc`, where
   they're handled by the test runner of rustbuild and the supplementary script
   `src/etc/htmldocck.py`.
 * Tests on search index generation are located in `src/test/rustdoc-js`, as a
   series of JavaScript files that encode queries on the standard library search
   index and expected results.
+* Tests on the "UI" of rustdoc (the terminal output it produces when run) are in
+  `src/tests/rustdoc-ui`
+* Tests on the "GUI" of rustdoc are in `src/tests/rustdoc-gui`. These use a
+  [NodeJS tool called
+  browser-UI-test](https://github.com/GuillaumeGomez/browser-UI-test/) that uses
+  puppeteer to run tests in a headless browser and check rendering and
+  interactivity.
+
+## Constraints
+
+We try to make rustdoc work reasonably well with JavaScript disabled, and when
+browsing local files. We support
+[these browsers](https://rust-lang.github.io/rfcs/1985-tiered-browser-support.html#supported-browsers).
+
+Supporting local files (`file:///` URLs) brings some surprising restrictions.
+Certain browser features that require secure origins, like `localStorage` and
+Service Workers, don't work reliably. We can still use such features but we
+should make sure pages are still usable without them.
+
+## Multiple runs, same output directory
+
+Rustdoc can be run multiple times for varying inputs, with its output set to the
+same directory. That's how cargo produces documentation for dependencies of the
+current crate. It can also be done manually if a user wants a big
+documentation bundle with all of the docs they care about.
+
+HTML is generated independently for each crate, but there is some cross-crate
+information that we update as we add crates to the output directory:
+
+ - crates.js holds a list of all crates in the output directory.
+ - search-index.js holds a list of all searchable items.
+ - For each trait, there is a file under `implementors/.../trait.TraitName.js`
+   containing a list of implementors of that trait. The implementors may be in
+   different crates than the trait, and the JS file is updated as we discover
+   new ones.
+
+## Use cases
+
+There are a few major use cases for rustdoc that you should keep in mind when
+working on it:
+
+### Standard library docs
+
+These are published at https://doc.rust-lang.org/std as part of the Rust release
+process. Stable releases are also uploaded to specific versioned URLs like
+https://doc.rust-lang.org/1.57.0/std/. Beta and nightly docs are published to
+https://doc.rust-lang.org/beta/std/ and https://doc.rust-lang.org/nightly/std/.
```

Comment:
I think none of these links will render as `<a>` tags. You need to add `<` and `>` around them.